torch/csrc/jit/python_custom_class.cpp

```diff
@@ -0,0 +1,71 @@
+#include <torch/csrc/jit/python_custom_class.h>
+#include <torch/csrc/jit/script/sugared_value.h>
+
+namespace torch {
+namespace jit {
+
+struct CustomMethodProxy;
+struct CustomObjectProxy;
+
+void initPythonCustomClassBindings(PyObject* module) {
+  auto m = py::handle(module).cast<py::module>();
+
+  m.def("_get_custom_class_python_wrapper", [](const std::string& str) {
+    auto cu = classCU();
+    c10::NamedTypePtr named_type =
+        cu->get_type("__torch__.torch.classes." + str);
+    if (!named_type || !named_type->cast<ClassType>()) {
+      std::stringstream err;
+      err << "Class " << str << " not registered!";
+      throw std::runtime_error(err.str());
+    }
+    c10::ClassTypePtr class_type = named_type->cast<ClassType>();
+    Function* ctor_method = class_type->getMethod("__init__");
+    if (!ctor_method) {
+      std::stringstream err;
+      err << "Class ";
+      if (auto name = class_type->name()) {
+        err << name->qualifiedName() << " ";
+      }
+      err << "does not have an __init__ method defined!";
+      throw std::runtime_error(err.str());
+    }
+
+    // Need to wrap __init__ in another function that actually returns the
+    // object so that torch.classes.Foo() doesn't just return None
+    auto wrapper_fn_name =
+        class_type->name()->qualifiedName() + ".__init__wrapper";
+    Function* ctor_wrapper;
+    if (classCU()->find_function(wrapper_fn_name)) {
+      ctor_wrapper = &classCU()->get_function(wrapper_fn_name);
+    } else {
+      auto graph = std::make_shared<Graph>();
+      ctor_wrapper = classCU()->create_function(wrapper_fn_name, graph);
```

Comment:
FunctionSchema is not registered so keyword argument handling for init will fail.